packages/manager/modules/manager-pci-common/src/api/hook/usePCIFeatureAvailability.ts

```diff
@@ -0,0 +1,30 @@
+import { useQuery } from '@tanstack/react-query';
+import { apiClient } from '@ovh-ux/manager-core-api';
+import { useApplicationName } from '@/hooks/useApplicationName/useApplicationName';
+
+type FeatureAvailabilityQueryParams = Partial<{
+  app: string;
+}>;
+
+const getFeatureAvailability = (
+  features: string[],
+  params?: FeatureAvailabilityQueryParams,
+) =>
+  apiClient.aapi.get(`/feature/${features.join(',')}/availability`, {
+    params,
+  });
+
+const useFeatureAvailability = <T extends string[]>(
```

Comment:
We could narrow the type as const :
```suggestion
const useFeatureAvailability = <T extends readonly string[]>(
```

Reply:
Why only this property ?

Reply:
Doing this will infer T as the union instead of simple strings